src/autocomplete/databases/mongo/mongo-extract-commands.ts

```diff
@@ -0,0 +1,267 @@
+import json5 from 'json5';
+import {
+    CollectionOperationContext,
+    FilterModifierContext,
+    FindMethodContext,
+    FindMethodModifierContext,
+    HintModifierContext,
+    InsertOneMethodContext,
+    LimitModifierContext,
+    MaxModifierContext,
+    MinModifierContext,
+    MongoParser,
+    ReturnKeyModifierContext,
+    ShowRecordIdModifierContext,
+    SkipModifierContext,
+    SortModifierContext,
+} from './generated/MongoParser';
+import {MongoParserVisitor} from './generated/MongoParserVisitor';
+import {ParserSyntaxError, SqlErrorListener, createParser} from '../../shared';
+import {MongoLexer} from './generated/MongoLexer';
+import {getParseTree} from './mongo-autocomplete';
+
+type FindModifier =
+    | {
+          method:
+              | 'skip'
+              | 'filter'
+              | 'min'
+              | 'max'
+              | 'returnKey'
+              | 'showRecordId'
+              | 'limit'
+              | 'hint';
+          parameters: unknown;
+      }
+    | {method: 'sort'; parameters?: unknown; options?: unknown};
+
+interface FindCommand {
+    method: 'find';
+    collectionName: string;
+    parameters?: unknown;
+    options?: unknown;
+    explain?: {
+        parameters?: unknown;
+    };
```

Comment:
aren't all of these strings or arrays? why don't we have typings for them?

Reply:
Just because it is user input which maybe anything, if we want to be sure is it string or array, we need to validate value